pkgs/development/compilers/gcc/5/default.nix

```diff
@@ -237,10 +237,16 @@ stdenv.mkDerivation ({
 
   # This should kill all the stdinc frameworks that gcc and friends like to
   # insert into default search paths.
-  prePatch = if stdenv.isDarwin then ''
+  prePatch = stdenv.lib.optionalString stdenv.isDarwin ''
```

Comment:
By making it into `optionalString` instead of `if ... then ... else null`, you actually change the hash on Linux 😄